bot/resources/tags/venv.md

```diff
@@ -0,0 +1,20 @@
+**Virtual Environments**
+
+Virtual environments are isolated Python environments, which make it easier to keep your system clean and manage dependencies. By default, when activated, only libraries and scripts installed in the virtual environment are accessible, preventing cross-project dependency conflicts, and allowing easy isolation of requirements.
+
+To create a new virtual environment, you can use the standard library `venv` module: `python3 -m venv .venv` (replace `python3` with `python` or `py` on Windows)
+
+Then, to activate the new virtual environment:
+
+**Windows** (PowerShell): `.venv\Scripts\Activate.ps1`
+or (Command Prompt): `.venv\Scripts\activate.bat`
+**MacOS / Linux** (Bash): `source .venv/bin/activate`
+
+Packages can then be installed to the virtual environment using `pip`, as normal.
+
+For more information, take a read of the [documentation](https://docs.python.org/3/library/venv.html). Your IDE, for example [VSCode](https://code.visualstudio.com/docs/python/environments#_select-and-activate-an-environment) or [PyCharm](https://www.jetbrains.com/help/pycharm/creating-virtual-environment.html), likely has `venv` integration.
```

Comment:
I think the bit about other IDEs should be reworded. I think it should be more clear that the links relate to the corresponding editors' virtual environment documentation rather than just information about the editors itself. I also think it should be specifically recommending looking at your editors venv documentation.

For example, it is common that a user may set up a venv using a guide similar to this, and then be confused when they click the green run button in the IDE and it runs with the system python rather than in the venv.

Something like:
```
If you run your code through your editor you should check it's docs so you can set it up to use your venv, for example, see the [VSCode](link) or [Pycharm](link) docs.
```
Thoughts?

Reply:
Yep, I struggled to keep that section short, while keeping it relevant. I think that sums it up nicely.

Just a slight reword:

> If you run code through your editor, check the relevant documentation on how to use your new virtual environment. For example, see the [VSCode](https://code.visualstudio.com/docs/python/environments#_select-and-activate-an-environment) or [PyCharm](https://www.jetbrains.com/help/pycharm/creating-virtual-environment.html) docs.

Is that alright?

Reply:
Yeah, that sounds great.

This is maybe slightly more concise again

```
If you run code through your editor, check it's documentation on how to make it use your virtual environment. For example, see the VSCode or PyCharm docs.
```

Anything along these lines or what you suggested is good though